File: profiling/profiler/utils/Transformations.cs
using System;

namespace profiler.utils
{
    public enum TransformationType
    {
        Rigid = 0,
        Affine = 1
    }
    class Transformations
    {
        public static float[] GetTransformation(TransformationType transformationType, float imageDimensionX, float imageDimensionY, float imageDimensionZ, float dx, float dy, float dz)
        {
            if (transformationType == TransformationType.Rigid)
                throw new NotImplementedException();

            if (transformationType == TransformationType.Affine)
            {
                return new[]
                {
                    imageDimensionX, 0,0, dx,
                    0,imageDimensionY,0, dy,
                    0,0,imageDimensionZ, dz,
                    0,0,0,1
                };
            }

            return null;
        }
    }
}

File: profiling/profiling_ui/transformations.cpp
#include "transformations.h"


Transformations::Transformations()
{
}


Transformations::~Transformations()
{
}

const char* Transformations::EnumToString(TransformationType t)
{
	switch (t)
	{
	case TransformationType::Rigid:
		return "Rigid";
	case TransformationType::Affine:
		return "Affine";
	}
}

void Transformations::GetTransformation(TransformationType transformationType, float imageDimensionX, float imageDimensionY, float imageDimensionZ, float dx, float dy, float dz, float *transformation)
{
	if (transformationType == TransformationType::Rigid)
		throw "error";

	if (transformationType == TransformationType::Affine)
	{
		*transformation = imageDimensionX;
		*++transformation;
		*transformation = 0;
		*++transformation;
		*transformation = 0;
		*++transformation;
		*transformation = dx;
		*++transformation;
		*transformation = 0;
		*++transformation;
		*transformation = imageDimensionY;
		*++transformation;
		*transformation = 0;
		*++transformation;
		*transformation = dy;
		*++transformation;
		*transformation = 0;
		*++transformation;
		*transformation = 0;
		*++transformation;
		*transformation = imageDimensionZ;
		*++transformation;
		*transformation = dz;
		*++transformation;
		*transformation = 0;
		*++transformation;
		*transformation = 0;
		*++transformation;
		*transformation = 0;
		*++transformation;
		*transformation = 1.0;
	}
}